shared/configs.ts

```diff
@@ -35,12 +35,21 @@ export function isGeneralConfig(config: FlatConfigItem) {
   return (!config.files && !config.ignores) || isIgnoreOnlyConfig(config)
 }
 
+/**
+ * Given a list of matched globs, if an unignore (leading !) is the last one, then the file no longer matches the glob set
+ */
+function filterUnignoreGlobs(globs: string[]) {
+  if (!globs.length || globs[globs.length - 1].startsWith('!'))
+    return []
```

Comment:
I am not sure if I follow, why would the unignore being the last makes the glob empty?

Reply:
I was primarily trying to make the change as small as possible within the existing code. Basically for the globalIgnore case...they are equivalent because if it's unignored, then we don't return globs anyways (i.e. the code carries on). It's by definition impossible to be ignored if the globs match as an unignore.

![image](https://github.com/user-attachments/assets/f2131e96-378a-47b5-a350-79c29dd38222)


However, I put up a different take I was thinking about that might be better for the per config case. It might be a little more/less of a brain twister. In the new code, we disconnect the pushing of the config from the pushing of the globs. This new case will then return the unignored globs, but not push the config, which may be more useful for debugging or more clear as shown below (i.e. it _does_ match technically)

Old approach:
![image](https://github.com/user-attachments/assets/ebae3d34-803e-445b-83f4-7eb1d19dba67)

Revised approach:
![image](https://github.com/user-attachments/assets/e98b9e3e-6ae5-41d7-afd1-1764a7092251)


In terms of why cueing on the last item works. Globs are evaluated in order, so you can walk through the list of globs for a config. Given `**/foo/special.js`:
**/*.js.  - File is ignored
!**/special.js. - File is unignored (this negates the previous matches)
**/foo/special.js - File is reignored (this negates the negation, or just readds it from a base state)

So using that logic, since we walk them in order, if the last matching glob is a positive matcher, than effectively all the other globs don't matter, it matches; the result is the file is ignored. If the last matching glob is an unignore (!), than no matter what the order of the things preceding it is, the final result is that it undoes all the ignores and it doesn't match.

Reply:
But won't there be the case of mixing of different types?

```ts
**/*.js
**/foo/*.js
!**/foo/special/*
**/foo/special/**/*.ts
...
```

I think we might better run a proper test for each glob with their priority?

Reply:
The globs we are cuing on[have already been matched on line 69](https://github.com/eslint/config-inspector/pull/109/files#diff-a10a94352a485a8610fb4a9c890ae472c75669158a1d39dd17e5059a1e10a791R69). So it's impossible to generate a file name that would cause both of these to match since they have mutually exclusive file extensions:
**/foo/*.js
**/foo/special/**/*.ts

Reply:
@antfu I think we're on slightly different schedules, but let me know if you want to hop on discord or something and take this off async delay!

Reply:
I am still not sure if I understand the logic here. Could we have some test cases to at least guard the behavior here?

Reply:
Done! There was no testing in this package previously so I also had to setup and configure all that, so it's expanded the PR a bit. I used Mocha and followed the general style from other ESLint packages so that hopefully it's familiar/consistent